File: producer.rb
# Requires that the mongo gem be installed on the system.
# Run command: ruby producer.rb

require 'mongo'
include Mongo

# connecting to the database
client = MongoClient.new("localhost", 27017, :pool_size => 1024)
db     = client['default']
coll   = db['buffer']

COUNT = 100000

# inserting documents
start = Time.new
COUNT.times { |i| coll.insert({ :event => 'fast', :timestamp => Time.new }) }

client.close

puts "total: " + COUNT.to_s
puts "msg/s: " + (COUNT.to_f / (Time.new - start).to_f).to_s

File: producer.py
#!/usr/bin/python

# Modified final source from http://shtylman.com/post/the-tail-of-mongodb/.
# Requires that pymongo be installed on the system.
# Run command: python producer.py

import time
import datetime
from pymongo import MongoClient

c = MongoClient(max_pool_size=1024)
#c.write_concern['w'] = 0
db = c.default
coll = db.buffer

COUNT = 100000

start = datetime.datetime.utcnow()
for i in range(0, COUNT):
    coll.insert({ "event": "fast", "timestamp": datetime.datetime.utcnow() })
    #time.sleep(0.0001)

c.close()

print("total: ", COUNT)
print("msg/s: ", COUNT/(datetime.datetime.utcnow() - start).total_seconds())
